### PR TITLE
[Modules] Log Analytics Workspace - added support for tables, dataExports, identity

### LINKED ISSUE #2640: [Feature Request]: Update API version, add tables and data exports to the Microsoft.OperationalInsights/workspaces module
### Description

- Update the API version of `Microsoft.OperationalInsights/workspaces` module to 2022-10-01
- add new child resources:
  - [dataExports](https://learn.microsoft.com/en-us/azure/templates/microsoft.operationalinsights/workspaces/dataexports?pivots=deployment-language-bicep)
  - [tables](https://learn.microsoft.com/en-us/azure/templates/microsoft.operationalinsights/workspaces/tables?pivots=deployment-language-bicep)

### PR DESCRIPTION
**Pull request by krbar**
# Description

This PR resolves #2640 and extends the Log Analytics Workspace module (`Microsoft.OperationalInsights/workspaces`). Following functionality has been added/updated:

- Update module's API versions
- Adding support for child resources:
  - [dataExports](https://learn.microsoft.com/en-us/azure/templates/microsoft.operationalinsights/workspaces/dataexports?pivots=deployment-language-bicep)
  - [tables](https://learn.microsoft.com/en-us/azure/templates/microsoft.operationalinsights/workspaces/tables?pivots=deployment-language-bicep)
- Adding support for a resource identity

## Pipeline references

| Pipeline |
| - |
| [OperationalInsights - Workspaces](https://github.com/Azure/ResourceModules/actions/workflows/ms.operationalinsights.workspaces.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/Microsoft.OperationalInsights/workspaces/dataExports/readme.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# OperationalInsights Workspaces DataExports `[Microsoft.OperationalInsights/workspaces/dataExports]`
+
+This module deploys OperationalInsights Workspaces DataExports.
+// TODO: Replace Resource and fill in description
```
Comment (ahmadabdalla) [inline modules/Microsoft.OperationalInsights/workspaces/dataExports/readme.md:4]: Please remove placeholder or add information

File: modules/Microsoft.OperationalInsights/workspaces/dataExports/readme.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+| :-- | :-- | :-- | :-- |
+| `destination` | object | `{object}` | Destination properties. |
+| `enable` | bool | `False` | Active when enabled. |
+| `enableDefaultTelemetry` | bool | `True` | Enable telemetry via the Customer Usage Attribution ID (GUID). |
+| `tableNames` | array | `[]` | An array of tables to export, for example: ['Heartbeat', 'SecurityEvent']. |
+
+
+### Parameter Usage: `<ParameterPlaceholder>`
+
+// TODO: Fill in Parameter usage
```
Comment (ahmadabdalla) [inline modules/Microsoft.OperationalInsights/workspaces/dataExports/readme.md:45]: Please remove placeholder or add parameter usage

File: modules/Microsoft.OperationalInsights/workspaces/deploy.bicep [hunk 3]
```diff
@@ -315,3 +362,6 @@ output logAnalyticsWorkspaceId string = logAnalyticsWorkspace.properties.custome
 
 @description('The location the resource was deployed into.')
 output location string = logAnalyticsWorkspace.location
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedPrincipalId string = systemAssignedIdentity && contains(logAnalyticsWorkspace.identity, 'principalId') ? logAnalyticsWorkspace.identity.principalId : ''
```
Comment (ahmadabdalla) [inline modules/Microsoft.OperationalInsights/workspaces/deploy.bicep:367]:
```suggestion
output systemAssignedIdentityPrincipalId string = systemAssignedIdentity && contains(logAnalyticsWorkspace.identity, 'principalId') ? logAnalyticsWorkspace.identity.principalId : ''
```

File: modules/Microsoft.OperationalInsights/workspaces/tables/readme.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+| `restoredLogs` | object | `{object}` |  | Restore parameters. |
+| `retentionInDays` | int | `-1` |  | The table retention in days, between 4 and 730. Setting this property to -1 will default to the workspace retention. |
+| `schema` | object | `{object}` |  | Table's schema. |
+| `searchResults` | object | `{object}` |  | Parameters of the search job that initiated this table. |
+| `totalRetentionInDays` | int | `-1` |  | The table total retention in days, between 4 and 2555. Setting this property to -1 will default to table retention. |
+
+
+### Parameter Usage: `<ParameterPlaceholder>`
+
+// TODO: Fill in Parameter usage
```
Comment (ahmadabdalla) [inline modules/Microsoft.OperationalInsights/workspaces/tables/readme.md:47]: Please remove or add parameter usage

File: modules/Microsoft.OperationalInsights/workspaces/dataExports/readme.md (hunk 2 above)
Comment (krbar) [reply modules/Microsoft.OperationalInsights/workspaces/dataExports/readme.md]: Updated

File: modules/Microsoft.OperationalInsights/workspaces/tables/readme.md (hunk 4 above)
Comment (krbar) [reply modules/Microsoft.OperationalInsights/workspaces/tables/readme.md]: Updated

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**